src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -259,25 +363,49 @@ out ImmutableArray<(BoundExpression, BoundDagTemp)> bindings)
             BoundDagTemp input,
             BoundConstantPattern constant,
             ArrayBuilder<BoundDagDecision> decisions,
-            ArrayBuilder<(BoundExpression, BoundDagTemp)> bindings,
-            ref HashSet<DiagnosticInfo> discardedUseSiteDiagnostics)
+            ArrayBuilder<(BoundExpression, BoundDagTemp)> bindings)
+        {
+            if (constant.ConstantValue == ConstantValue.Null)
+            {
+                decisions.Add(new BoundNullValueDecision(constant.Syntax, input));
+            }
+            else
+            {
+                if (input.Type.CanContainNull())
+                {
+                    decisions.Add(new BoundNonNullDecision(constant.Syntax, input));
+                }
+
+                var convertedInput = ConvertToType(input, constant.Syntax, constant.Value.Type, decisions);
+                decisions.Add(new BoundNonNullValueDecision(constant.Syntax, constant.ConstantValue, convertedInput));
+            }
+        }
+
+        /// <summary>
+        /// This can be used instead of Debug.Assert as it more reliably breaks to the debugger
+        /// when an assertion fails (it is unaffected by exception filters on enclosing frames).
+        /// </summary>
+        [Conditional("DEBUG")]
+        private void Assert(bool condition, string message = null)
```

Comment:
>void [](start = 16, length = 4)

`static`

Reply:
Addressed in [https://github.com/dotnet/roslyn/pull/25662](https://github.com/dotnet/roslyn/pull/25662)

---
In reply to: [175846649](https://github.com/dotnet/roslyn/pull/25544#discussion_r175846649) [](ancestors = 175846649)